elastic/beats
# Rename google-pubsub Filebeat input to gcp-pubsub

## What does this PR do?

This PR is to rename `google-pubsub` input to `gcp-pubsub` input. This is a part of the renaming process to change `googlecloud` to `gcp` in both Metricbeat and Filebeat. 

Warning log message is also added when user is using the old input name:
```
2020-10-28T13:54:08.841-0600    WARN    [gcp.pubsub]    gcppubsub/input.go:79   google-pubsub input name is deprecated, please use gcp-pubsub instead   {"pubsub_project": "elastic-observability", "pubsub_topic": "my-gcp-pubsub-topic-name", "pubsub_subscription": {"Name":"my-gcp-pubsub-subscription-name","NumGoroutines":1,"MaxOutstandingMessages":1000,"Create":true}}
```

## Checklist




## How to test this PR locally

This should not be a breaking change so both `google-pubsub` and `gcp-pubsub` input names should work.
1. Test `google-pubsub` name:
```
filebeat.inputs:
- type: google-pubsub
  enabled: true
  project_id: elastic-observability
  topic: my-gcp-pubsub-topic-name
  subscription.name: my-gcp-pubsub-subscription-name
  credentials_file: "/Users/kaiyansheng/Desktop/elastic-observability-5537b831234a.json"
```

1. Test `gcp-pubsub` name:
```
filebeat.inputs:
- type: gcp-pubsub
  enabled: true
  project_id: elastic-observability
  topic: my-gcp-pubsub-topic-name
  subscription.name: my-gcp-pubsub-subscription-name
  credentials_file: "/Users/kaiyansheng/Desktop/elastic-observability-5537b831234a.json"
```

## Related issues

- Relates https://github.com/elastic/beats/issues/19467
